tests/test_simple.py

```diff
@@ -3,102 +3,110 @@
-    )
+def test_model(gpu_lock):
+    with gpu_lock():
+        # A baseline config (single-gpu, bf16, flash-attn).
+        # Also tests for multiple data loaders.
+        run_test_script(
+            f"test_{TEST_MODEL}", CONFIG_COMMON + ["training.num_workers=2"], compare=f"test_{TEST_MODEL}_safe"
+        )
 
 
 @pytest.mark.slow
 @pytest.mark.depends(on=["test_model"])
-def test_model_dp2():
+def test_model_dp2(gpu_lock):
     # Simple data-parallel.
-    run_test_script(f"test_{TEST_MODEL}_dp2", CONFIG_COMMON, num_gpus=2, compare=f"test_{TEST_MODEL}")
+    with gpu_lock(num_gpus=2):
+        run_test_script(f"test_{TEST_MODEL}_dp2", CONFIG_COMMON, num_gpus=2, compare=f"test_{TEST_MODEL}")
 
 
 @pytest.mark.slow
-def test_model_dp2_timeout():
+def test_model_dp2_timeout(gpu_lock):
     # Test sampling timeout
     # TODO: Find a better way to test this
-    run_test_script(
-        f"test_{TEST_MODEL}_dp2",
-        CONFIG_COMMON
-        + [
-            # Use a short timeout
-            "model.distributed.timeout=4",
-            # Make a dataset that would timeout under the distributed timeout
-            'data.datasets.training={"type":"test_slow"}',
-            "data.datasets.training.type=test_slow",
-            "data.datasets.training.sleep=6",
-            # Use a bigger timeout for the dataset.
-            "training.timeout=10",
-            # Remove testing clutter.
-            f"model.multi_stage.debug_param_init=0",
-            f"model.multi_stage.debug_layer_outputs=0",
-            f"model.multi_stage.debug_layer_gradients=0",
-            f"model.multi_stage.debug_all_param_gradients=0",
-        ],
-        num_gpus=2,
-    )
+    with gpu_lock(num_gpus=2):
+        run_test_script(
+            f"test_{TEST_MODEL}_dp2",
+            CONFIG_COMMON
+            + [
+                # Use a short timeout
+                "model.distributed.timeout=4",
+                # Make a dataset that would timeout under the distributed timeout
+                'data.datasets.training={"type":"test_slow"}',
+                "data.datasets.training.type=test_slow",
+                "data.datasets.training.sleep=6",
+                # Use a bigger timeout for the dataset.
+                "training.timeout=10",
+                # Remove testing clutter.
+                f"model.multi_stage.debug_param_init=0",
+                f"model.multi_stage.debug_layer_outputs=0",
+                f"model.multi_stage.debug_layer_gradients=0",
+                f"model.multi_stage.debug_all_param_gradients=0",
+            ],
+            num_gpus=2,
+        )
 
 
 @pytest.mark.slow
 @pytest.mark.depends(on=["test_model"])
-def test_model_tp2():
+def test_model_tp2(gpu_lock):
     # Simple tensor-parallel.
-    run_test_script(
-        f"test_{TEST_MODEL}_tp2",
-        CONFIG_COMMON + ["model.distributed.tensor_parallel=2"],
-        num_gpus=2,
-        compare=f"test_{TEST_MODEL}",
-    )
+    with gpu_lock(num_gpus=2):
```

Comment:
Ideally I'd use a mark or a better fixture to avoid this. Any pytest expert here knows how to do it? (@bigximik @tscholak)